src/Microsoft.Diagnostics.Monitoring/DiagnosticServices.cs

```diff
@@ -39,71 +63,84 @@ public IEnumerable<int> GetProcesses()
 
         public async Task<OperationResult<Stream>> GetDump(int? pid, DumpType mode)
         {
-            if (!pid.HasValue)
-            {
-                pid = GetSingleProcessId();
-            }
+            int pidValue = GetSingleProcessId(pid);
 
-            string dumpFilePath = FormattableString.Invariant($@"{Path.GetTempPath()}{Path.DirectorySeparatorChar}{Guid.NewGuid()}_{pid.Value}");
+            string dumpFilePath = FormattableString.Invariant($@"{Path.GetTempPath()}{Path.DirectorySeparatorChar}{Guid.NewGuid()}_{pidValue}");
             NETCore.Client.DumpType dumpType = MapDumpType(mode);
 
             if (RuntimeInformation.IsOSPlatform(OSPlatform.Windows))
             {
                 // Get the process
-                Process process = Process.GetProcessById(pid.Value);
+                Process process = Process.GetProcessById(pidValue);
                 await Dumper.CollectDumpAsync(process, dumpFilePath, dumpType);
             }
             else
             {
                 await Task.Run(() =>
                 {
-                    var client = new DiagnosticsClient(pid.Value);
+                    var client = new DiagnosticsClient(pidValue);
                     client.WriteDump(dumpType, dumpFilePath);
                 });
             }
 
-            return CreateResult(pid.Value, new AutoDeleteFileStream(dumpFilePath));
+            return CreateResult<Stream>(pidValue, new AutoDeleteFileStream(dumpFilePath));
         }
 
-        public Task<OperationResult<Stream>> StartCpuTrace(int? pid, int durationSeconds, CancellationToken cancellationToken)
+        public async Task<OperationResult<IStreamResult>> StartCpuTrace(int? pid, int durationSeconds, CancellationToken cancellationToken)
         {
             if ((durationSeconds < 1) || (durationSeconds > MaxTraceSeconds))
             {
                 throw new InvalidOperationException("Invalid duration");
             }
 
-            if (!pid.HasValue)
+            int pidValue = GetSingleProcessId(pid);
+
+            DiagnosticsMonitor monitor = new DiagnosticsMonitor(new CpuProfileConfiguration());
```

Comment:
This should be disposed once the streaming operation completes.

Reply:
I see that you are already doing that by requiring the caller to dispose the IStreamResult.